src/main/java/de/zalando/aruha/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -0,0 +1,125 @@
+        this.subscriptionStreamerFactory = subscriptionStreamerFactory;
+        this.jsonMapper = objectMapper;
+    }
+
+    private class SubscriptionOutputImpl implements SubscriptionOutput {
+        private boolean headersSent;
+        private final HttpServletResponse response;
+        private final OutputStream out;
+
+        SubscriptionOutputImpl(final HttpServletResponse response, final OutputStream out) {
+            this.response = response;
+            this.out = out;
+            this.headersSent = false;
+        }
+
+        @Override
+        public void onInitialized() throws IOException {
+            if (!headersSent) {
+                headersSent = true;
+                response.setStatus(HttpStatus.OK.value());
+                response.setContentType("application/x-json-stream");
+                out.flush();
+            }
+        }
+
+        @Override
+        public void onException(final Exception ex) {
+            LOG.warn("Exception occurred while streaming", ex);
+            if (!headersSent) {
+                headersSent = true;
+                try {
+                    if (ex instanceof NakadiException) {
+                        writeProblemResponse(Response.Status.BAD_REQUEST, ((NakadiException) ex).getProblemMessage());
+                    } else {
+                        writeProblemResponse(Response.Status.SERVICE_UNAVAILABLE, "Failed to continue streaming");
+                    }
+                } catch (final IOException e) {
+                    LOG.error("Failed to write exception to response", e);
+                }
+            } else {
+                LOG.warn("Exception found while streaming, but no data could be provided to client", ex);
+            }
+        }
+
+        void writeProblemResponse(final Response.StatusType statusCode, final String message) throws IOException {
+            response.setStatus(statusCode.getStatusCode());
+            response.setContentType("application/problem+json");
+            jsonMapper.writer().writeValue(out, Problem.valueOf(statusCode, message));
+        }
+
+        @Override
+        public void streamData(final byte[] data) throws IOException {
+            headersSent = true;
+            out.write(data);
+            out.flush();
+        }
+    }
+
+    @RequestMapping(value = "/subscriptions/{subscription_id}/events", method = RequestMethod.GET)
+    public StreamingResponseBody streamEvents(
+            @PathVariable("subscription_id") final String subscriptionId,
+            @RequestParam(value = "window_size", required = false, defaultValue = "100") final int windowSize,
+            @RequestParam(value = "commit_timeout", required = false, defaultValue = "30") final int commitTimeout,
+            @RequestParam(value = "batch_limit", required = false, defaultValue = "1") final int batchLimit,
```

Comment:
I thought, that values, that are not specified will be nulls, and I don't want to check them on `0`.
So everything that has default value is primitive and everything that doesn't have it and could be omitted - object.